cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/TimePartitionedFileSetArguments.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib;
+
+import co.cask.cdap.api.annotation.Beta;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Helpers for manipulating runtime arguments of time-partitioned file sets.
+ */
+@Beta
+public class TimePartitionedFileSetArguments {
+
+  public static final String OUTPUT_PARTITION_TIME = "output.partition.time";
+  public static final String INPUT_START_TIME = "input.start.time";
+  public static final String INPUT_END_TIME = "input.end.time";
+
+  /**
+   * Set the time of the output partition when using TimePartitionedFileSet as an OutputFormatProvider.
+   * This time is used as the partition key for the new file, and also to generate an output file path - if that path
+   * is not explicitly given as an argument itself.
+   * @param time The time in milli seconds.
+   */
+  public static void setOutputPartitionTime(Map<String, String> arguments, long time) {
+    arguments.put(OUTPUT_PARTITION_TIME, Long.toString(time));
```

Comment:
It does, and this is the only way I could figure out how to do it with the current APIs:
- you use this method to set the property, and call getDataset()
- under the covers, that calls TPFSDefinition.getDataset(). That uses the time of this property to set the output path property of the underlying fileset. That is a file name.
- Both FileSet and TimePartitionedFileSet have a way to get the arguments, so a client can find out what was configured.

You are right that it is possible to bypass this by setting the output path explicitly in the FileSet output arguments. Then it will write to the path, but still use the time configured with this method to register the partition (actually that code is not written, but conceptually, it would do that). 

Or maybe I don't understand exactly what you mean...